packages/ruleset/src/utils/logger-factory.js

```diff
@@ -0,0 +1,91 @@
+const log = require('loglevel');
+const minimatch = require('minimatch');
+
+/**
+ * This class serves as a factory for creating loggers implemented by the 'loglevel' package.
+ * The primary benefit provided by this factory class is the ability to honor logging-level-related
+ * command-line options for loggers that haven't yet been created.
+ */
+module.exports = class LoggerFactory {
+  static _factory = null;
+
+  constructor() {
+    this.rootLogger = log;
+    this.loggerSettings = [];
+  }
+
+  /**
+   * @returns a single instance of our factory.
+   */
+  static newInstance() {
+    if (!this._factory) {
+      this._factory = new LoggerFactory();
+    }
+    return this._factory;
+  }
+
+  /**
+   * A logger setting consists of a logger name (a glob-like string,
+   * e.g. 'my-*-logger' or 'my-first-logger') and a log level
+   * (e.g. 'error', 'warn', 'info', 'debug', or 'trace').
+   * This function adds the logger setting specified by "name" and
+   * "logLevel" to the current array of logger settings.
+   * @param {*} name the name of the logger (glob-like string)
+   * @param {*} logLevel the log level to set on the logger
+   */
+  addLoggerSetting(name, logLevel) {
```

Comment:
One last thought - should we have any validation logic that ensures the user provides a supported level (either here or in run-validator.js) ?

I don't think it's necessary, just thinking out loud

Reply:
yeah, I'll add some validation logic here.